src/Microsoft.OData.Core/ODataResource.cs

```diff
@@ -414,13 +413,18 @@ public void AddFunction(ODataFunction function)
             }
         }
 
-        private static void VerifyProperties(IEnumerable<ODataProperty> properties)
+        private static void VerifyProperties(IEnumerable<ODataPropertyInfo> properties)
         {
             if (properties != null)
             {
                 ODataCollectionValue collection = null;
-                foreach (var property in properties)
+                foreach (ODataPropertyInfo propertyInfo in properties)
                 {
+                    if (!(propertyInfo is ODataProperty property))

```

Comment:
can we use " is not " key word? Otherwise, can we update the Lang version since it's new targeting version.

Reply:
Fixed